### DISCUSSION TITLE
Setting background of GIF image sequence to be transparent

### DISCUSSION THREAD

**Discussion by Jacherr** [ImageMagick/ImageMagick#3481] [Help] [answered]
Hi!

I'm trying to create a GIF out of a single PNG image. What I'm doing is cloning the image, then rotating it 45 degrees using `-distort SRT 1,45`. This leaves a sequence of two images: the first one has no rotation, and the second one has a 45 degree rotation.

I disabled virtual pixel so there is some background visible in the rotated image. By default, the background of the rotated image will be the first image in the sequence, because I don't set a disposal method using `-set dispose`. if I do `-set dispose background`, the background becomes black. Using `-background transparent` with `-set dispose background` does not seem to work-  the background doesn't become transparent. Is there any way to change the background of the image to be transparent, instead of either being the previous image or completely black?

The full command I'm using is `magick rose.png -virtual-pixel none -set dispose background \( -clone 0 -distort srt 1,45 \)`. This command produces a black/brown background for [this input file](https://legacy.imagemagick.org/Usage/basics/rose.png) instead of the transparent background I'm after.

I'm using ImageMagick version 7.0.11-3 on Ubuntu 20.04.

Any help would be appreciated.

**Comment by snibgo** [comment]
You want the first frame to show through the transparent parts of the second frame. Is that correct? If so, then remove "-set dispose background".

**Reply by Jacherr** [reply]
That's not what I'm after, I want the transparent parts of the second frame to be transparent - i.e. with an opacity of zero. Attempting to do this using `-set dispose background` and `-background transparent` doesn't seem to work - the disposal seems to be overriding the `-background` option regardless of order.

**Reply by Jacherr** [reply]
[This](https://cdn.discordapp.com/attachments/819916774283149322/826779116975816704/attachment.gif) is an example of the output with the script `magick rose.png -set dispose background -background transparent -virtual-pixel none \( -clone 0 -distort 1,45 \)`. The background parts of the second image aren't transparent here - they're a solid, non-transparent colour. Removing `-set dispose background` causes the first frame to show through the second frame - also not a transparent background.

**Comment by snibgo** [comment, marked as answer]
Ah, okay. "-set dispose background" is correct, but the default background colour is white, so you need to set that. This works for me, v7.0.8-64:
```
magick -delay 50 rose: -virtual-pixel none -background None -set dispose background \( -clone 0 -distort srt 1,45 \) x.gif
```
![x](https://user-images.githubusercontent.com/33812368/113140295-23132c80-9220-11eb-947f-283b89a7992d.gif)

**Reply by Jacherr** [reply]
Hmm, that's strange.. It appears that every browser I tried to view this GIF on didn't display the transparency of the second frame, instead opting to fill it with a different colour - usually white. That might have been the issue all along. Not quite sure what the cause of that is, though.

**Comment by snibgo** [comment]
Showing the GIF I created on a simple coloured web page...
```
<html><head>
</head><body bgcolor=#88f>
<img src="x.gif" />
</body></html>
```
... with Firefox v87.0, the corners of the second frame show the expected blue colour.

Other web browsers may vary, but this shows the GIF has transparent corners when the second frame is shown. (These are "transparent black", so browsers that ignore transparency will show this as black.)
